### PR TITLE
Run GoLintCLI in CI

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Also removed unnecessary GOPATH stuff.

**Comment by jhendrixMSFT** [comment]
@richardpark-msft this one is ready to go now.

File: fuzz_test.go [hunk 1]
```diff
@@ -192,8 +194,8 @@ func fuzzUnmarshal(data []byte) int {
 	}
 
 	for _, t := range types {
-		unmarshal(&buffer{b: data}, t)
-		readAny(&buffer{b: data})
+		_ = unmarshal(&buffer{b: data}, t)
```
Comment (jhendrixMSFT) [inline fuzz_test.go:197]: This got me looking closer at these fuzz tests.  While they exercise code-paths, there's nothing that actually triggers the tests to fail.  I opened https://github.com/Azure/go-amqp/issues/53 to track this.

File: session.go [hunk 2]
```diff
@@ -385,7 +395,10 @@ func (s *Session) mux(remoteBegin *performBegin) {
 				s.muxFrameToLink(link, fr.body)
 
 			case *performEnd:
-				s.txFrame(&performEnd{}, nil)
+				if err := s.txFrame(&performEnd{}, nil); err != nil {
+					s.err = err
```
Comment (serbrech) [inline session.go:399]: what's the consequence of setting the session err ?
when is it checked, and where is it returned?

Comment (serbrech) [reply session.go:399]: previously, it was only ever set when calling Close() it seems.

Comment (serbrech) [reply session.go:399]: is failing to send 1 frame a reason to set the error on the session?

Comment (jhendrixMSFT) [reply session.go:399]: A fair question.  What happens if we fail to send the frame, are we asked to send again?  The endpoint doesn't care?  It seems dubious to just ignore the error (clearly the frame is important else we wouldn't send it).  Makes me wonder if things like this are part of the flakiness we see with connections getting dropped, but it's just speculation.

Comment (jhendrixMSFT) [reply session.go:399]: Digging into this more, `txFrame()` will only return an error if the connection has been closed, and presumably we fail someplace else early on.  So, I think these errors can be safely ignored given the current implementation (the lack of comments explaining this is beyond frustrating).

**Review by richardpark-msft — APPROVED** [review]
Looks good.

**Pull request merged**